coderd/workspaceagents.go

```diff
@@ -1703,12 +1703,14 @@ func (api *API) workspaceAgentReportStats(rw http.ResponseWriter, r *http.Reques
 		return nil
 	})
 	errGroup.Go(func() error {
-		err := api.Database.UpdateWorkspaceLastUsedAt(ctx, database.UpdateWorkspaceLastUsedAtParams{
-			ID:         workspace.ID,
-			LastUsedAt: now,
-		})
-		if err != nil {
-			return xerrors.Errorf("can't update workspace LastUsedAt: %w", err)
+		if req.ConnectionCount > 0 {
```

Comment:
Is it some kind of regression you found?

Reply:
(Side-note: It's less overhead if this `if` statement was outside the `errGroup.Go`.)

While I think this behavior is correct(!), I want to point out that due to how stats are counted, this won't behave as expected.

Simply opening the workspace dashboard page will increase the connection count since coder is connecting to the agent (e.g. figuring out ports?)

```
coder=# select * from workspace_agent_stats where connection_count > 0 order by created_at desc limit 3;
-[ RECORD 1 ]------------------+-------------------------------------
id                             | 52537501-eeba-432a-8283-348c0e93ea86
created_at                     | 2023-11-21 10:42:04.885102+00
user_id                        | 022e411a-256d-40ab-b6c5-4f0a998f3a73
agent_id                       | e10744dd-c11d-473b-859d-146ef04f0457
workspace_id                   | a2085528-004e-468f-953a-a0fc0fe6210c
template_id                    | def19fea-ece5-4abb-82aa-8a6fb32ecc80
connections_by_proto           | {"TCP": 5}
connection_count               | 5
rx_packets                     | 30
rx_bytes                       | 3025
tx_packets                     | 25
tx_bytes                       | 3120
connection_median_latency_ms   | 0.431
session_count_vscode           | 0
session_count_jetbrains        | 0
session_count_reconnecting_pty | 0
session_count_ssh              | 0
-[ RECORD 2 ]------------------+-------------------------------------
id                             | edd94fe5-2887-47d3-bac8-548ac079cbb8
created_at                     | 2023-11-21 10:41:34.881285+00
user_id                        | 022e411a-256d-40ab-b6c5-4f0a998f3a73
agent_id                       | e10744dd-c11d-473b-859d-146ef04f0457
workspace_id                   | a2085528-004e-468f-953a-a0fc0fe6210c
template_id                    | def19fea-ece5-4abb-82aa-8a6fb32ecc80
connections_by_proto           | {"TCP": 3}
connection_count               | 3
rx_packets                     | 18
rx_bytes                       | 1815
tx_packets                     | 15
tx_bytes                       | 1872
connection_median_latency_ms   | 0.356
session_count_vscode           | 0
session_count_jetbrains        | 0
session_count_reconnecting_pty | 0
session_count_ssh              | 0
```

I'm not sure what the fix _should_ be, but one way to fix it here and now is to also require a session count in any of the four categories.

Perhaps it's also worth a new issue to track reporting improvements on the agent.

Reply:
| Is it some kind of regression you found?

Basically we use `last_used_at` for determining if a workspace is eligible for dormancy. If the field is updated every time stats are reported regardless of user connections then it'll never be considered inactive if it's using autostart.